apps/jobbrowser/src/jobbrowser/apis/hive_query_api.py

```diff
@@ -114,21 +115,40 @@ def app(self, appid):
 
     return app
 
-  def action(self, appid, action):
-    message = {'message': '', 'status': 0}
+  def action(self, query_ids, action, request=None):
+    message = {'actions': {}, 'status': 0}
 
     if action.get('action') == 'kill':
-      for queryid in appid:
-        notebook = {}
-        snippet = {'result': {'handle': {'secret': queryid, 'guid': queryid, 'has_result_set': False}}}
-        connector_id = 'hive'
+      for query_id in query_ids:
+        action_details = {}
+
+        try:
+          self.kill_query(query_id, request)
+          action_details['status'] = 0
+          action_details['message'] = _('kill action performed')
+        except Exception as ex:
+          LOG.error(ex)
+          message['status'] = -1
+          action_details['status'] = -1
+          action_details['message'] = _('kill action failed')
```

Comment:
We might want to provide the error to the end user if possible

Reply:
Thanks, done.